operator/pkg/lbipam/lbipam.go

```diff
@@ -28,19 +29,20 @@ import (
 	cilium_client_v2alpha1 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
 	"github.com/cilium/cilium/pkg/k8s/resource"
 	slim_core_v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	slim_meta "github.com/cilium/cilium/pkg/k8s/slim/k8s/apimachinery/pkg/api/meta"
 	slim_meta_v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
 	client_typed_v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/typed/core/v1"
 	"github.com/cilium/cilium/pkg/option"
 )
 
 const (
 	// The condition added to services to indicate if a request for IPs could be satisfied or not
-	ciliumSvcRequestSatisfiedCondition = "io.cilium/lb-ipam-request-satisfied"
+	ciliumSvcRequestSatisfiedCondition = "cilium.io/IPAMRequestSatisfied"
 
-	ciliumPoolIPsTotalCondition     = "io.cilium/ips-total"
-	ciliumPoolIPsAvailableCondition = "io.cilium/ips-available"
-	ciliumPoolIPsUsedCondition      = "io.cilium/ips-used"
-	ciliumPoolConflict              = "io.cilium/conflict"
+	ciliumPoolIPsTotalCondition     = "cilium.io/IPsTotal"
+	ciliumPoolIPsAvailableCondition = "cilium.io/IPsAvailable"
+	ciliumPoolIPsUsedCondition      = "cilium.io/IPsUsed"
+	ciliumPoolConflict              = "cilium.io/PoolConflict"
```

Comment:
Just our of curiosity, is there a k8s standard or guide somewhere which describes naming convention for fields such as these. I can for the live of me not remember why I chose the java package (reverse dns) style here. But doing the kebab-case for the rest seems quite normal https://kubernetes.io/docs/reference/labels-annotations-taints/

Reply:
You mentioned it in [the issue](https://github.com/cilium/cilium/issues/24945#issuecomment-1560626174), actually. It's what the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1623-standardize-conditions/README.md?plain=1#L81) has as an example, and condition types on regular resources like pods and deployments are camel cased as well.

This could be moved to a different PR if you want, since it's not as important as fixing the bug that breaks strategic merge patching.

Reply:
Ah! I had missed that part. Thank you for pointing that out and fixing this issue :pray: . I think these can stay, given its for conformance to the KEP and these fields are mostly for internal use anyway.